creusot/tests/should_succeed/type_invariants/generated.mlcfg

```diff
@@ -198,7 +196,367 @@ module Generated_UseFoo
+  predicate postcondition_once [#"../generated.rs" 32 12 32 19] (self : generated_testclosure_closure0) (_ : ()) (result : Generated_Sum10_Type.t_sum10)
+    
+   =
+    [#"../generated.rs" 1 0 1 0] true
+  val generated_TestClosure_Closure0 [#"../generated.rs" 32 12 32 19] (_1 : generated_testclosure_closure0) : Generated_Sum10_Type.t_sum10
+    requires {[#"../generated.rs" 1 0 1 0] Inv0.inv _1}
+    ensures { [#"../generated.rs" 32 12 32 19] Inv1.inv result }
+    
+end
+module Generated_TestClosure_Closure0
+  use export Generated_TestClosure_Closure0_Type
+  use Generated_Sum10_Type as Generated_Sum10_Type
+  clone Generated_Impl0_Invariant as Invariant0
+  clone CreusotContracts_Invariant_Inv_Interface as Inv1 with
+    type t = Generated_Sum10_Type.t_sum10
+  clone Generated_Sum10_Type_Inv as Generated_Sum10_Type_Inv0 with
+    predicate Inv0.inv = Inv1.inv,
+    predicate Invariant0.invariant' = Invariant0.invariant',
+    axiom .
+  clone CreusotContracts_Invariant_Inv_Interface as Inv0 with
+    type t = generated_testclosure_closure0
+  let function field_0 [#"../generated.rs" 32 12 32 19] (self : generated_testclosure_closure0) : Generated_Sum10_Type.t_sum10
+    
+   = [@vc:do_not_keep_trace] [@vc:sp]
+    [#"../generated.rs" 1 0 1 0] let Generated_TestClosure_Closure0 a = self in a
+  predicate resolve [#"../generated.rs" 32 12 32 19] (_1 : generated_testclosure_closure0) =
+    [#"../generated.rs" 1 0 1 0] true
+  predicate precondition [#"../generated.rs" 32 12 32 19] (self : generated_testclosure_closure0) (_ : ()) =
+    [#"../generated.rs" 1 0 1 0] true
+  predicate postcondition_once [#"../generated.rs" 32 12 32 19] (self : generated_testclosure_closure0) (_ : ()) (result : Generated_Sum10_Type.t_sum10)
+    
+   =
+    [#"../generated.rs" 1 0 1 0] true
+  let rec cfg generated_TestClosure_Closure0 [#"../generated.rs" 32 12 32 19] [@cfg:stackify] [@cfg:subregion_analysis] (_1 : generated_testclosure_closure0) : Generated_Sum10_Type.t_sum10
+    requires {[#"../generated.rs" 1 0 1 0] Inv0.inv _1}
+    ensures { [#"../generated.rs" 32 12 32 19] Inv1.inv result }
+    
+   = [@vc:do_not_keep_trace] [@vc:sp]
+  var _0 : Generated_Sum10_Type.t_sum10;
+  var _1 : generated_testclosure_closure0 = _1;
+  {
+    goto BB0
+  }
+  BB0 {
+    _0 <- field_0 _1;
+    _1 <- (let Generated_TestClosure_Closure0 a = _1 in Generated_TestClosure_Closure0 (any Generated_Sum10_Type.t_sum10));
+    assert { [@expl:type invariant] Inv0.inv _1 };
+    assume { resolve _1 };
+    return _0
+  }
+  
+end
+module Generated_TestClosure_Closure0_Inv
+  use Generated_Sum10_Type as Generated_Sum10_Type
+  clone CreusotContracts_Invariant_Inv_Stub as Inv1 with
+    type t = Generated_Sum10_Type.t_sum10
+  clone CreusotContracts_Invariant_Inv_Stub as Inv0 with
+    type t = Closure00.generated_testclosure_closure0
+  clone Generated_TestClosure_Closure0_Interface as Closure00 with
+    predicate Inv0.inv = Inv0.inv,
+    predicate Inv1.inv = Inv1.inv
```

Comment:
There is a cyclic dependency here: `Inv0` requires the closure type exported by `Closure00` which in turn requires the invariant symbol. One solution would be to first generate a `use Generated_TestClosure_Closure0_Type`.